src/plugins/screenshot_mode/common/context.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+// **PLEASE NOTE**
+// The functionality in this file targets a browser environment AND is intended to be used both in public and server.
+// For instance, reporting uses these functions when starting puppeteer to set the current browser into "screenshot" mode.
+
+type Context = Record<string, unknown>;
+
+declare global {
+  interface Window {
+    __KBN_SCREENSHOT_CONTEXT__?: Context;
+  }
+}
```

Comment:
Why extend window? Feels like this should be plugin state and using a global like this feels inappropriate.

Reply:
- It is [invoked](https://github.com/puppeteer/puppeteer/blob/main/docs/api.md#pageevaluateonnewdocumentpagefunction-args) by the puppeteer right after the document is created. There is no Kibana context at this point, so as anything else.
- We cannot access plugins because we cannot use any imports/references. The function should be fully [serializable](https://github.com/puppeteer/puppeteer/blob/a858cf7021e9bb4da2373e32a1d83ead62e161e7/src/common/helper.ts#L185).

Reply:
Ah, looked around for something like that but didn't find it. Makes sense.